format/spec.md

```diff
@@ -1761,6 +1764,10 @@ The reference Java implementation uses a type 4 uuid and XORs the 4 most signifi
 
 Java writes `-1` for "no current snapshot" with V1 and V2 tables and considers this equivalent to omitted or `null`. This has never been formalized in the spec, but for compatibility, other implementations can accept `-1` as `null`. Java will no longer write `-1` and will use `null` for "no current snapshot" for all tables with a version greater than or equal to V3.
 
+### Legacy naming for GZIP compressed Metadata JSON files
+
+Some implementations have written GZIP compressed metadata JSON files with the suffix `metadata.json.gz`. The reference Java implementation will interpret files with this naming convention as GZIP files for backwards compatibility.
```

Comment:
Isn't this backwards of what is suggested in the mailing list discussion?  `.gz.metadata.json` is the legacy format and `.metadata.json.gz` is the proposed representation?

Reply:
I commented on https://github.com/apache/iceberg/pull/12598#discussion_r2067028228 I think people want to retroactively change a decision that has been canonicalized in Iceberg for quite some time.

Reply:
I think this is the only place where filenames should be mentioned, but this note doesn't seem sufficient to me. First, we should document the behavior expected by Hadoop/FS tables that rely on a predictable file name. Second, this is where we should document that although there are no requirements for naming, the convention is encouraged to be `<unique-name>.<compression-ext>.metadata.json`. Then note that some implementations have put the compression extension at the end, like `<unique-name>.metadata.json.gz`.

Reply:
I think we are a bit to ambiguous here in the spec right now or at least have not been specific in what we are leaving up to an implementation.

For example in our description of metadata tables it definitely seems like we are mandating a postfix metadata.json. It also feels like we are specifying other parts like the version and the random bits.

 https://github.com/apache/iceberg/blob/main/format/spec.md?plain=1#L1065 ? 

I'd prefer we either specifically say we think file names end in xxx or we say that actual choice of file names is up to the implementation.

Reply:
Lets consolidate discussion on https://github.com/apache/iceberg/pull/12598#discussion_r2070444650 thread